### DISCUSSION TITLE
[Question] Is this web extension respect privacy if I use it in Tor-Browser???

### DISCUSSION THREAD

**Discussion by amalgame21** [gildas-lormeau/SingleFile#1118] [Q&A]
I noticed that the 'cli' version it has a `--browser-args` flag which seems can be used to specify socks5 proxy in tor.

Is using this plugin in Tor-Browser automatically do it?

Also, if I use it in Tor-Browser, will it leak some of the anonymous information to the internet?

Thanks

**Comment by gildas-lormeau** [comment]
From my point of view, it should be fine to use SingleFile in Tor-Browser regarding the proxy issue. If an extension was able to circumvent the proxy without doing it on purpose, that would be a bug in Tor-Browser.
SingleFile can be detected. As far as I know, only this information can leak on the websites you are browsing. It means that a website (or an embedded third-party resource like an ad) might be able to detect that you used SingleFile to save a page on that site and when you did it. It would have to be done on purpose though. So far, I have never heard of this kind of "attack", which remains quite minor IMHO.

**Reply by amalgame21** [reply]
Thanks for your answer!
Since you mentioned they might be able to detect I use SingleFile, I am wondering if I just enable the extension and do not save the page by pressing the button, are they still have a chance to detect it?
If yes, I may just keep the extension disable before using it.

**Reply by gildas-lormeau** [reply]
Unfortunately yes, because of some limitations in the Web Extension API. For example, this code will display a log in the console if SingleFile is simply enabled:
```js
addEventListener("single-file-new-font-face", () => console.log("SingleFile detected"));
new FontFace("", "");
```

Of course, disabling SingleFile fortunately fixes this problem.

**Reply by amalgame21** [reply]
Thanks for your advice!!!!
Learn a lot!!!

**Comment by gildas-lormeau** [comment]
Since it's a question, I'm moving the issue in the Discussions tab.
